### PR TITLE
chore(demo): update Taiga UI to `v4`

### PR DESCRIPTION
**Pull request by nsbarsukov**

### CODE REVIEW

File: projects/demo/src/pages/documentation/changelog/changelog.style.less [hunk 1]
```diff
@@ -1,42 +1,25 @@
...
-        margin-right: 0.5rem;
+//@import '@taiga-ui/addon-doc/styles/index';
+
+.t-changelog {
+    /*
+    TODO: uncomment usage of mixins after this PR is merged:
+    // https://github.com/taiga-family/taiga-ui/pull/8108
+    */
+    //.markdown();
+    //.changelog-styles();
```
Comment (nsbarsukov) [inline projects/demo/src/pages/documentation/changelog/changelog.style.less:9]: Fix before merge

Comment (nsbarsukov) [reply projects/demo/src/pages/documentation/changelog/changelog.style.less:9]: * https://github.com/taiga-family/taiga-ui/pull/8108

Comment (nsbarsukov) [reply projects/demo/src/pages/documentation/changelog/changelog.style.less]: Done ✅

File: projects/demo/src/app/app.component.ts [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import {ChangeDetectionStrategy, Component} from '@angular/core';
 import {DemoPath} from '@demo/constants';
-import {TUI_SANITIZER} from '@taiga-ui/core';
+import {TUI_SANITIZER} from '@taiga-ui/legacy';
```
Comment (waterplea) [inline projects/demo/src/app/app.component.ts:3]: We do not need this anymore at all

File: projects/demo/src/app/app.routes.ts [hunk 3]
```diff
@@ -32,228 +29,176 @@ export const appRoutes: Routes = [
...
-            title: 'Stackblitz Starter',
-        },
-    },
-    {
-        path: '**',
-        redirectTo: DemoPath.WhatIsMaskito,
+        title: 'Stackblitz Starter',
     },
-];
+].map(tuiProvideRoutePageTab);
```
Comment (waterplea) [inline projects/demo/src/app/app.routes.ts:188]: Smart!

File: projects/demo/src/app/modules/example-primary-tabs-icons/angular-logo.component.ts [hunk 4]
```diff
@@ -1,11 +1,11 @@
...
+import {TuiSvgComponent} from '@taiga-ui/legacy';
+import {PolymorpheusComponent} from '@taiga-ui/polymorpheus';
 
 @Component({
     standalone: true,
     selector: 'angular-logo',
-    imports: [TuiSvgModule],
+    imports: [TuiSvgComponent],
     template: `
         <tui-svg src="assets/icons/angular.svg"></tui-svg>
```
Comment (waterplea) [inline projects/demo/src/app/modules/example-primary-tabs-icons/angular-logo.component.ts:10]: Changer it to `img`, get rid of `tui-svg` everywhere to drop `TUI_SANITIZER` and dompurify

File: projects/demo/src/pages/documentation/what-is-maskito/what-is-maskito.component.ts [hunk 5]
```diff
@@ -1,20 +1,15 @@
 import {ChangeDetectionStrategy, Component} from '@angular/core';
 import {RouterLink} from '@angular/router';
 import {DemoPath} from '@demo/constants';
-import {TuiAddonDocModule} from '@taiga-ui/addon-doc';
-import {TuiLinkModule} from '@taiga-ui/core';
-import {TuiIslandModule, TuiMarkerIconModule} from '@taiga-ui/kit';
+import {TuiAddonDoc} from '@taiga-ui/addon-doc';
+import {TuiLink} from '@taiga-ui/core';
+import {TuiAvatar} from '@taiga-ui/kit';
+import {TuiIslandDirective} from '@taiga-ui/legacy';
```
Comment (waterplea) [inline projects/demo/src/pages/documentation/what-is-maskito/what-is-maskito.component.ts:7]: Maybe switch to card?

**Review by waterplea — APPROVED** [review]

File: projects/demo/src/app/app.component.ts (hunk 2 above)
Comment (nsbarsukov) [reply projects/demo/src/app/app.component.ts]: Deleted in https://github.com/taiga-family/maskito/pull/1390/commits/6868980272983af084bcbf3a522945d95e3b4dd5

File: projects/demo/src/app/modules/example-primary-tabs-icons/angular-logo.component.ts (hunk 4 above)
Comment (nsbarsukov) [reply projects/demo/src/app/modules/example-primary-tabs-icons/angular-logo.component.ts]: Done in https://github.com/taiga-family/maskito/pull/1390/commits/cb38567fbed4c02079bff4493fc3f70ff69b983a

File: projects/demo/src/pages/documentation/what-is-maskito/what-is-maskito.component.ts (hunk 5 above)
Comment (nsbarsukov) [reply projects/demo/src/pages/documentation/what-is-maskito/what-is-maskito.component.ts]: Done in https://github.com/taiga-family/maskito/pull/1390/commits/55a7b0932b92b96e5ef3434e94acb6f8a0f1defe

**Pull request merged**